test/translator/request_sense_test.cc

```diff
@@ -0,0 +1,116 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lib/translator/request_sense.h"
+
+#include "absl/base/casts.h"
+#include "gtest/gtest.h"
+
+// Tests
+
+namespace {
+TEST(RequestSense, ToNvmeSuccess) {
+  const scsi::RequestSenseCommand request_sense_cmd = {
+      .allocation_length = 100,
+      .control_byte = {.naca = 0},
+  };
+  uint32_t allocation_length = 0;
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&request_sense_cmd);
+  absl::Span<const uint8_t> scsi_cmd =
+      absl::MakeSpan(ptr, sizeof(scsi::RequestSenseCommand));
+  EXPECT_EQ(translator::RequestSenseToNvme(scsi_cmd, allocation_length),
+            translator::StatusCode::kSuccess);
+  EXPECT_EQ(allocation_length, request_sense_cmd.allocation_length);
+}
+
+TEST(RequestSense, ToNvmeBadBuffer) {
+  const scsi::RequestSenseCommand request_sense_cmd = {
+      .allocation_length = 100,
+      .control_byte = {.naca = 0},
+
+  };
+  uint32_t allocation_length = 0;
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&request_sense_cmd);
+  absl::Span<const uint8_t> scsi_cmd = absl::MakeSpan(ptr, sizeof(1));
+  EXPECT_EQ(translator::RequestSenseToNvme(scsi_cmd, allocation_length),
+            translator::StatusCode::kInvalidInput);
+  EXPECT_EQ(allocation_length, 0);
+}
+
+TEST(RequestSense, ToScsiBadBuffer) {
+  const scsi::RequestSenseCommand request_sense_cmd{};
+  uint8_t buf[100];
+  absl::Span<uint8_t> buffer = buf;
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&request_sense_cmd);
+  absl::Span<const uint8_t> scsi_cmd = absl::MakeSpan(ptr, sizeof(1));
+  EXPECT_EQ(translator::RequestSenseToScsi(scsi_cmd, buffer),
+            translator::StatusCode::kInvalidInput);
+}
+
+TEST(RequestSense, ToNvmeBadControlByteNaca) {
+  const scsi::RequestSenseCommand request_sense_cmd = {
+      .allocation_length = 100,
+      .control_byte = {.naca = 1},
+
+  };
+  uint32_t allocation_length = 0;
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&request_sense_cmd);
+  absl::Span<const uint8_t> scsi_cmd =
+      absl::MakeSpan(ptr, sizeof(scsi::RequestSenseCommand));
+  EXPECT_EQ(translator::RequestSenseToNvme(scsi_cmd, allocation_length),
+            translator::StatusCode::kInvalidInput);
+  EXPECT_EQ(allocation_length, 0);
+}
+
+TEST(RequestSense, ToScsiDescriptor) {
+  uint8_t buf[100];
+  absl::Span<uint8_t> buffer = buf;
+  const scsi::RequestSenseCommand request_sense_cmd = {
+      .desc = 1, .allocation_length = 100, .control_byte = {.naca = 0}};
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&request_sense_cmd);
+  absl::Span<const uint8_t> scsi_cmd =
+      absl::MakeSpan(ptr, sizeof(scsi::RequestSenseCommand));
+  translator::StatusCode status =
+      translator::RequestSenseToScsi(scsi_cmd, buffer);
+  EXPECT_EQ(status, translator::StatusCode::kSuccess);
+
+  scsi::DescriptorFormatSenseData result{};
+  EXPECT_TRUE(translator::ReadValue(buf, result));
```

Comment:
Use `ASSERT_TRUE` for `ReadValue` if you're going to inspect the result, because inspecting the result doesn't make sense if `ReadValue` fails, so the test should not continue past that point.